lib/view_component/test_helpers.rb

```diff
@@ -5,11 +5,20 @@
 module ViewComponent
   module TestHelpers
     begin
+      require "capybara/dsl"
       require "capybara/minitest"
+
+      include Capybara::DSL
       include Capybara::Minitest::Assertions
 
+      def self.included(mod)
+        Capybara::Node::Simple.send(:define_method, :to_capybara_node) do
```

Comment:
The reason for the `to_capybara_node` method is to be able to pass nodes, i.e. what's returned by `find`, to the `within` method. The implementation of `within` expects the argument to respond to `to_capybara_node`. In a full blown browser based test the nodes are of type `Capybara::Node::Base`, which implements `to_capybara_node`. In this implementation the nodes are of type `Capybara::Node::Simple`, which does **not** implement `to_capybara_node`. The `Capybara::Node::Simple` class is intended to be used for text when no browser is involved. To quote the documentation:

> It is useful in that it does not require a session, an application or a driver, but can still use Capybara's finders and matchers on any string that contains HTML.

The implementation of the `within` method which I've included here is copied from Capybara. An alternative would be to check if the argument passed to the `within` method is of type `Capybara::Node::Simple`, not sure if that's preferred. I don't know if there are any other types that can be passed to `within` which responds to `to_capybara_node`. Most of the code in this PR is copied from Capybara, I deliberately tried to keep it as close to the original code to make it easier to incorporate any future changes in Capybara.

Reply:
> The implementation of the `within` method which I've included here is copied from Capybara. An alternative would be to check if the argument passed to the `within` method is of type `Capybara::Node::Simple`, not sure if that's preferred.

I think we should avoid modifying `Capybara::Node::Simple`. If capybara decides to implement that method or another library decides to copy this approach I believe we could end up with two gems that now have a compatibility issue due to modifying external dependencies, which we should strive to avoid.